core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -3401,11 +3410,36 @@ public PhysicalOperation visitTableExecute(TableExecuteNode node, LocalExecution
             return new PhysicalOperation(operatorFactory, outputMapping.buildOrThrow(), context, source);
         }
 
-        private int getWriterCount(Session session, Optional<PartitioningScheme> partitioningScheme, PlanNode source)
+        private int getWriterCount(Session session, Optional<PartitioningScheme> partitioningScheme, Optional<PartitioningScheme> preferredPartitioningScheme, PlanNode source)
         {
+            if (isSingleGatheringExchange(source)) {
```

Comment:
was this a bug? separate commit if so

Reply:
No, this is something we have to handle as part of this change. Basically, if you have a single distribution, you don't know where it came from (scaling partitioned or unpart) and which writer count to use. Previously it was easier since we only used a partitioning scheme to determine the correct writer count.